### PR TITLE
return errors in json when json format

### LINKED ISSUE #14: Return error message(s?) via JSON for vim-go
Split out of #1 

From @fatih:

```go
func foo() error {
	bar()

	// no return yet as I'm still developting this
}

func bar() error {
	return errors.New("foo")
}
```

When I put my cursor on top of `bar()` and call `:GoExpandErr` it returns an error saying:

```
2017/12/28 17:58:43 ignoring type-checking error: demo.go:54:1: missing return
{"start":51,"end":51,"lines":["\tif err := bar(); err != nil {","\t\treturn err","\t}"]}
```

It actually returns something. However this is wrong. The error should be part of json and we should be able to decide whether we can change the buffer or not. I think the error message above can be mitigated, but not sure how expanderr implemented it.

### PR DESCRIPTION
**Pull request by wsny**
Fixes #14

### CODE REVIEW

File: expanderr.go [hunk 1]
```diff
@@ -349,7 +349,7 @@ func (e *expansion) typeCheck(pkgname string, files []*ast.File) error {
 	conf := types.Config{
 		Importer: defaultImporter(),
 		Error: func(err error) {
-			log.Printf("ignoring type-checking error: %v", err)
+			// log.Printf("ignoring type-checking error: %v", err)
```
Comment (stapelberg) [inline expanderr.go]: Why are you commenting out this line?

If errors should be swallowed here, that deserves an explanatory comment.

File: expanderr.go [hunk 2]
```diff
@@ -758,6 +758,18 @@ func main() {
...
 	if err := logic(o, &build.Default, posn, *noErrReturnStr); err != nil {
+		if *formatFlag == "json" {
+			jsonErr := json.NewEncoder(o).Encode(struct {
+				Error string `json:"error"`
+			}{
+				Error: err.Error(),
+			})
+			if jsonErr != nil {
+				log.Fatal(err)
+				log.Fatal(jsonErr)
```
Comment (stapelberg) [inline expanderr.go:776]: This line is unreached due to the log.Fatal(err) above.

Why are you introducing jsonErr instead of just using err?

**Review by stapelberg — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: expanderr.go (hunk 1 above)
Comment (wsny) [reply expanderr.go]: As far as I can tell this is just a debug log. It is printed out but then everything keeps going and nothing is done with the fact that there was an error here.

File: expanderr.go (hunk 2 above)
Comment (wsny) [reply expanderr.go:776]: Ah, true. Figured if this got hit it might be useful to see both what the error was and why its not able to be encoded.

Comment (stapelberg) [reply expanderr.go:776]: In practice, encoding won’t fail, so this is largely a philosophical question :)

In that case, you’d need to log.Print one, then log.Fatal on the other error.

File: expanderr.go (hunk 1 above)
Comment (stapelberg) [reply expanderr.go]: When I wrote that line, I wasn’t sure whether ignoring type-checking errors is the correct thing to do. I’m still not sure, so I figure we should tell users what we’re doing.

Comment (wsny) [reply expanderr.go]: At first I was starting to try and accumulate them as warnings and return those at the end before just discarding them as I did here. I'll see if I can get that working later.

**Comment by wsny** [comment]
"I'll see if I can get that working later." - wsny

I guess later is 3 years.

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
Thanks!

**Pull request merged**